cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/JmxTransReconciler.java

```diff
@@ -4,149 +4,85 @@
  */
 package io.strimzi.operator.cluster.operator.assembly;
 
-import io.fabric8.kubernetes.api.model.ConfigMap;
-import io.fabric8.kubernetes.api.model.LocalObjectReference;
-import io.fabric8.kubernetes.api.model.apps.Deployment;
 import io.strimzi.api.kafka.model.JmxTransResources;
-import io.strimzi.api.kafka.model.Kafka;
-import io.strimzi.operator.cluster.ClusterOperatorConfig;
-import io.strimzi.operator.cluster.model.ImagePullPolicy;
-import io.strimzi.operator.cluster.model.JmxTrans;
 import io.strimzi.operator.cluster.operator.resource.ResourceOperatorSupplier;
-import io.strimzi.operator.common.Annotations;
 import io.strimzi.operator.common.Reconciliation;
-import io.strimzi.operator.common.Util;
 import io.strimzi.operator.common.operator.resource.ConfigMapOperator;
 import io.strimzi.operator.common.operator.resource.DeploymentOperator;
 import io.strimzi.operator.common.operator.resource.ServiceAccountOperator;
 import io.vertx.core.Future;
 
-import java.util.List;
-
 /**
- * Class used for reconciliation of JMX Trans deployment. This class contains both the steps of the JMX Trans
- * reconciliation pipeline and is also used to store the state between them.
+ * JMX Trans is not supported anymore. The JmxTransReconciler class is still used to delete the JMXTrans deployment in
+ * case it was used by a previous Strimzi version. The only functionality here is now deleting the JMXTrans deployment.
+ * This will be completely removed in Strimzi 0.40.0.
```

Comment:
Looking at the code I was wondering if we should log a WARN message to make clear the user what's going to happen and why JMXTrans related resources were deleted.

Reply:
The deprecation of the API already logs warning messages about this. It also adds it to the `.status` section of the custom resource. I think that is enough.

Reply:
It logs deprecation which was ok until now. Not sure it's enough when the operator does an actual deletion of stuff if JMX is there. Unless the deprecation warning already mentions that JMX stuff is going to be deleted.

Reply:
AFAIU, right now the JMX reconciler is going to delete the JMX related stuff right? How about logging it's deleting the stuff. Will this code then be removed in the future, I guess? So no JMX deletion nor logging anymore? Or the reconciliation will be still there?

Reply:
That is covered in the proposal -> 0.35-0.39 will be deleting any existing JMX Trans deployment. From 0.40, it iwll be completely removed and we will not do anything (even delete it if it still exists).

Reply:
> That is covered in the proposal -> 0.35-0.39 will be deleting any existing JMX Trans deployment.

Imho, we should log (until 0.39 release) that JMX Trans are deleted if there is any. Then we'll remove all the code.
The current `JMXTrans is deprecated and removed in Strimzi 0.35.0` is pretty clear about API deprecation and support removed in 0.35.0 (so you cannot deploy JMX Trans anymore), but it isn't clear that it's going to remove/distrupt your JMX stuff (which instead will stay after 0.40.0 even if it still exists).

Reply:
But when do you want it to be logged? That is what is not clear to me. All the time?

> The current JMXTrans is deprecated and removed in Strimzi 0.35.0 is pretty clear about API deprecation and support removed in 0.35.0 (so you cannot deploy JMX Trans anymore), but it isn't clear that it's going to remove/distrupt your JMX stuff (which instead will stay after 0.40.0 even if it still exists).

TBH, the reality is that when you see any of the warnings, the deployment will be gone. So it will be IMHO pretty clear what `removed` means.

Reply:
Can we just make it clearer with updating the warning messages as ...

> JMXTrans is deprecated and related resources removed in Strimzi 0.35.0

?

Reply:
Ok, should be changed now.

Reply:
Thanks Jakub!